crates/proto/src/quic/quic_client_stream.rs

```diff
@@ -236,28 +287,30 @@ impl QuicClientStreamBuilder {
         name_server: SocketAddr,
         dns_name: String,
     ) -> Result<QuicClientStream, ProtoError> {
-        // ensure the ALPN protocol is set correctly
-        let mut crypto_config = self.crypto_config;
-        if crypto_config.alpn_protocols.is_empty() {
-            crypto_config.alpn_protocols = vec![quic_stream::DOQ_ALPN.to_vec()];
-        }
-        let early_data_enabled = crypto_config.enable_early_data;
+        if let Config::Partly {
+            crypto_config,
+            transport_config,
+            ..
+        } = self.0
+        {
+            // ensure the ALPN protocol is set correctly
+            let mut crypto_config = crypto_config;
+            if crypto_config.alpn_protocols.is_empty() {
+                crypto_config.alpn_protocols = vec![quic_stream::DOQ_ALPN.to_vec()];
+            }
 
-        let mut client_config = ClientConfig::new(Arc::new(crypto_config));
-        client_config.transport_config(self.transport_config.clone());
+            let mut client_config = ClientConfig::new(Arc::new(crypto_config));
+            client_config.transport_config(transport_config.clone());
 
-        endpoint.set_default_client_config(client_config);
+            endpoint.set_default_client_config(client_config);
+        }
 
         let connecting = endpoint.connect(name_server, &dns_name)?;
         // TODO: for Client/Dynamic update, don't use RTT, for queries, do use it.
 
-        let quic_connection = if early_data_enabled {
-            match connecting.into_0rtt() {
-                Ok((new_connection, _)) => new_connection,
-                Err(connecting) => connecting.await?,
-            }
-        } else {
-            connecting.await?
+        let quic_connection = match connecting.into_0rtt() {
```

Comment:
Currently removed the if check, see https://github.com/bluejekyll/trust-dns/issues/2002#issuecomment-1687069256.

Reply:
Resolved as described in https://github.com/bluejekyll/trust-dns/issues/2002#issuecomment-1687714407.